jwst/wfss_contam/wfss_contam.py

```diff
@@ -0,0 +1,88 @@
+#
+#  Top level module for WFSS contamination correction.
+#
+from os.path import splitext
+
+from jwst import datamodels
+from .observations import observation
+from ..lib.suffix import replace_suffix
+
+import logging
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+
+def contam_corr(input_model, waverange):
+    """
+    The main WFSS contam correction function
+
+    Parameters
+    ----------
+    input_model : `~jwst.datamodels.MultiSlitModel`
+        Input data model containing 2D spectral cutouts
+    waverange : `~jwst.datamodels.WavelengthrangeModel`
+        Wavelength range reference file model
+
+    Returns
+    -------
+    output_model : `~jwst.datamodels.MultiSlitModel`
+      A copy of the input_model that has been corrected
+
+    """
+    output_model = input_model.copy()
+
+    # Get the segmentation map for this grism exposure
+    seg_model = datamodels.open(input_model.meta.segmentation_map)
+
+    # Get the direct image from which the segmentation was constructed
+    direct_file = input_model.meta.direct_image
+    image_names = [direct_file]
+
+    # Get the grism WCS from the input model
+    grism_wcs = input_model.slits[0].meta.wcs
+    filter = input_model.meta.instrument.filter
+
+    # Create a simulated grism image containing all of the sources
+    # defined in the segmentation map
+    obs = observation(image_names, seg_model, grism_wcs, waverange, filter,
```

Comment:
This will create a simulated image always using the WCS of the first slit. Is this correct? Should the observation be slit specific, i.e. in the loop over slits?

Reply:
It only needs the overall grism image WCS info to simulate spectra for all sources and since the WCS stored in each slit instance is identical, except for the offset in origin for each slit, it can use any of the slit instances to get the WCS info. This particular call does in fact build a full-frame image containing simulated spectra for all sources. Later on in the step, that's used along with the spectrum for each individual source to correct each individual slit instance.

Reply:
Yes, you are right, I forgot wfss mode has a simple wcs.